test/models/filter_test.rb

```diff
@@ -241,4 +241,11 @@ class FilterTest < ActiveSupport::TestCase
     f.enforce_inherited_taxonomies
     assert_equal "(organization_id ^ (#{f.organizations.first.id}))", f.taxonomy_search
   end
+
+  test 'enforce_inherited_taxonomies does not build empty taxonomy search string' do
+    f = FactoryBot.build(:filter, :resource_type => 'Domain')
+    f.role = FactoryBot.build(:role, :organizations => [FactoryBot.build(:organization)])
+    f.save
+    assert_nil f.taxonomy_search
```

Comment:
Previously this would have allowed an empty string and I don't think that was expected

Reply:
see ^^